### DISCUSSION TITLE
Why is my onSubmit not working in PrimeReact Dialog?

### DISCUSSION THREAD

**Discussion by Simsopheap** [primefaces/community#732] [PrimeReact] [answered]
I see the Dialog renders `renderFooter `outside of form, Now how to solve it?

`  const footerContent = (
    <div>
      <Button label={translate("NAV.CLOSE")} icon="pi pi-times" onClick={() => setVisible(false)} className="p-button-text" />
      <Button type="submit" label={translate("NAV.SAVE")} icon="pi pi-save" className='mr-0' />
    </div>
  );`

`<Dialog header={header} draggable={false} visible={visible} style={{ width: '35rem', minWidth: '35rem' }} onHide={() => setVisible(false)} footer={footerContent}><form onSubmit={(e) => handleSubmit(e)}>....</form></Dialog>`

**Comment by melloware** [comment, marked as answer]
I answered on Stack Overflow please upvote: https://stackoverflow.com/a/71727194/502366
